### PR TITLE
[ISSUE-21] Class responses for push and push multicast (send/send all)

### PR DESCRIPTION
**Pull request by ndmytro**
* `push` method now returns a `FcmPushResponse` object, that has a fully compatible interface with SendResponse object of the Google's firebase-admin-python package.
 * `push_multicast` method now returns a `FcmPushMulticastResponse` object, that has a fully compatible interface with BatchResponse object of the Google's firebase-admin-python package.
 * The aforementioned methods may still rise exceptions when assembling the message for the request.
 * A bunch of exceptions has been added.

### CODE REVIEW

**Comment by akalex** [comment]
LGTM

File: async_firebase/messages.py [hunk 1]
```diff
@@ -215,3 +215,45 @@ class PushNotification:
...
+    @property
+    def success_count(self):
+        return self._success_count
+
+    @property
+    def failure_count(self):
+        return len(self.responses) - self.success_count
+
+
+class FcmPushResponse:
```
Comment (akalex) [inline async_firebase/messages.py]: I guess two new classes should be considered `dataclass`. What do you think?

File: async_firebase/messages.py [hunk 2]
```diff
@@ -215,3 +214,54 @@ class PushNotification:
...
+        :param fcm_response: a dictionary with the data that FCM returns as a payload
+        :param exception: an exception that may happen when communicating with FCM
+        """
+        self.message_id = fcm_response.get("name") if fcm_response else None
+        self.exception = exception
+
+    @property
+    def success(self) -> bool:
+        """A boolean indicating if the request was successful."""
+        return self.message_id is not None and not self.exception
```
Comment (akalex) [inline async_firebase/messages.py:238]: This double-check seems redundant you may leave only on condition.

File: async_firebase/messages.py (hunk 1 above)
Comment (ndmytro) [reply async_firebase/messages.py]: Let's think about it later

File: async_firebase/messages.py (hunk 2 above)
Comment (ndmytro) [reply async_firebase/messages.py:238]: Actually, the object may be constructed incorrectly, and in this case `success` attribute value will indicate this.

**Comment by akalex** [comment]
LGTM

**Review by akalex — APPROVED** [review]

**Pull request merged**